### ISSUE TITLE
Separate OpenMM ForceField template and parameter generation

### ISSUE THREAD

**Issue by pgrinaway** [choderalab/openmoltools#187]
As discussed [here](https://github.com/choderalab/perses/pull/83), it would be useful to either separate parameter and template generation, or store templates for efficiency. Unfortunately, it seems we have partial charges in the template, which complicates this and might require storing the template in the ffxml.

**Comment by jchodera** [comment]
I don't think we want to do this.  What we really need is a way to take a list of `molecules` and produce an `ffxml` file that defines both templates and any additional parameters.

**Comment by jchodera** [comment]
There is no meaningful way to separate residue templates and the additional `parmchk2`-generated parameters at this point.

**Comment by bas-rustenburg** [comment]
For me, I add extra atoms and bonds (dummies) after the antechamber and parmchk2 parameters, so I will need separate templates. I can't generate parameters for unphysical atoms, unless I missed an option to mark atoms as dummies to antechamber and openmoltools.

**Comment by jchodera** [comment]
> For me, I add extra atoms and bonds (dummies) after the antechamber and parmchk2 parameters, so I will need separate templates. I can't generate parameters for unphysical atoms, unless I missed an option to mark atoms as dummies to antechamber and openmoltools.

You will actually need to generate the `ffxml` files yourself since you will also be adding additional information about titration states, relative energies, etc.  I don't think we want that to go into `openmoltools` for now.

Instead, you will want to use `openmoltools` to generate residue templates (and extra parameters) for a set of protonation/tautomer states, and then you will want to convert these to your modified `ffxml` format with additional dummy atoms yourself.

**Comment by bas-rustenburg** [comment]
I'm confused, how did we previously generate templates? I'm already modifying  ffxml files, generated for each state,  but I assumed there was a template generator somewhere.

**Comment by bas-rustenburg** [comment]
Also, how do I then convert a template to an ffxml file, like you said?

**Comment by bas-rustenburg** [comment]
Sorry,  the forcefield code is just so confusing. I'm not even sure I need templates if I already have ffxml files.

**Comment by jchodera** [comment]
Let me go through this again with you in person tomorrow.

**Comment by bas-rustenburg** [comment]
Sounds good.
